src/common/hooks/account/use-switch-account.ts

```diff
@@ -0,0 +1,33 @@
+import { useWallet } from '@common/hooks/use-wallet';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import { hasSwitchedAccountsState, transactionAccountIndexState } from '@store/accounts';
+import { useCallback } from 'react';
+import { transactionNetworkVersionState } from '@store/transactions';
+
+const TIMEOUT = 350;
+
+export const useSwitchAccount = (callback?: () => void) => {
+  const { wallet, currentAccountIndex, doSwitchAccount } = useWallet();
+  const txIndex = useRecoilValue(transactionAccountIndexState);
+  const transactionVersion = useRecoilValue(transactionNetworkVersionState);
+  const [hasSwitched, setHasSwitched] = useRecoilState(hasSwitchedAccountsState);
+
+  const handleSwitchAccount = useCallback(
+    async index => {
+      if (typeof txIndex === 'number') setHasSwitched(true);
+      await doSwitchAccount(index);
+      if (callback) {
+        window.setTimeout(() => {
```

Comment:
Need an effect to handle cancelling timeout?

Reply:
we had one previously that produced errors, will look into this for a future improvement